### PR TITLE
File syncing using drag and drop

### PR DESCRIPTION
**Pull request by kylecombes**
Adds the ability to sync files between HydroShare and JupyterHub by dragging and dropping files between the HydroShare and JupyterHub file panes.

Current functionality:
* Frontend sends copy requests to backend when file is dragged from one pane and dropped in another
* Frontend sends move requests to backend when file is dragged from one folder in JupyterHub and dropped in another folder in JupyterHub or from one HydroShare folder to another
* Backend uploads files from JupyterHub to HydroShare. Uploading folders and their contents is buggy.
* Backend moves files around within JupyterHub

The UI is still pretty rough, but this needs to be merged with dev so the backend folks can work on the backend components.

### CODE REVIEW

File: backend/resource2.py [hunk 1]
```diff
@@ -35,10 +40,8 @@ def __init__(self, res_id, resource_handler):
         self.remote_folder = RemoteFolder(self.hs, self.res_id)
         self.local_folder = LocalFolder()
 
-        self.path_prefix = self.output_folder + "/" + self.res_id + "/" + self.res_id + "/data/contents/"
+        self.path_prefix = Path(self.output_folder) / self.res_id / self.res_id / 'data' / 'contents'
```
Comment (vickymmcd) [inline backend/resource2.py]: Do you not need quotes around the slashes?

File: backend/resource2.py [hunk 2]
```diff
@@ -300,21 +303,33 @@ def overwrite_JH_with_file_from_HS(self, filepath):
...
+            file_path = str(file_path)[1:]
+        full_file_path_rel_resource_root = Path(file_path)
+        file_extension = full_file_path_rel_resource_root.suffix
+        path_without_extension = str(full_file_path_rel_resource_root)[:-len(file_extension)]
+        # Drop the leading . from the file extension
+        file_extension = file_extension[1:]
+        if self.is_file_or_folder_in_HS(path_without_extension, file_extension):
+            try:
+                self.delete_file_or_folder_from_HS(full_file_path_rel_resource_root)
+            except hs_restclient.HydroShareHTTPException as e:
```
Comment (vickymmcd) [inline backend/resource2.py]: what does the hydroshare exception trigger?

File: backend/resource2.py (hunk 1 above)
Comment (kylecombes) [reply backend/resource2.py]: Not with pathlib. The `/` operator is overridden to join path components (so long as the first component is a `pathlib.Path`).

File: backend/resource2.py (hunk 2 above)
Comment (kylecombes) [reply backend/resource2.py]: Actually I should remove those `try...except` blocks.

**Pull request merged**